runtime/binding-tcp/src/main/java/io/aklivity/zilla/runtime/binding/tcp/internal/stream/TcpServerFactory.java

```diff
@@ -150,56 +152,28 @@ public MessageConsumer newStream(
     public void attach(
         BindingConfig binding)
     {
-        TcpBindingConfig tcpBinding = new TcpBindingConfig(binding);
-        router.attach(tcpBinding);
+        TcpBindingConfig tcpBinding = new TcpBindingConfig(binding, event);
+        bindings.put(binding.id, tcpBinding);
     }
 
     @Override
     public void detach(
         long bindingId)
     {
-        router.detach(bindingId);
-    }
-
-    private int handleAccept(
-        PollerKey acceptKey)
-    {
-        try
-        {
-            TcpBindingConfig binding = (TcpBindingConfig) acceptKey.attachment();
-            TcpOptionsConfig options = binding.options;
-
-            ServerSocketChannel server = (ServerSocketChannel) acceptKey.channel();
-
-            for (SocketChannel channel = router.accept(server); channel != null; channel = router.accept(server))
-            {
-                channel.configureBlocking(false);
-                channel.setOption(TCP_NODELAY, options.nodelay);
-                channel.setOption(SO_KEEPALIVE, options.keepalive);
-
-                InetSocketAddress local = (InetSocketAddress) channel.getLocalAddress();
-
-                onAccepted(binding, channel, local);
-            }
-        }
-        catch (Exception ex)
-        {
-            LangUtil.rethrowUnchecked(ex);
-        }
-
-        return 1;
+        bindings.remove(bindingId);
     }
 
-    private void onAccepted(
-        TcpBindingConfig binding,
+    public void onAccepted(
+        long bindingId,
         SocketChannel network,
         InetSocketAddress local)
     {
+        TcpBindingConfig binding = bindings.get(bindingId);
         final TcpRouteConfig route = binding.resolve(local);
 
         if (route != null)
         {
-            final TcpServer server = new TcpServer(binding.id, route.id, network);
+            final TcpServer server = new TcpServer(bindingId, route.id, network);
```

Comment:
```suggestion
            final TcpServer server = new TcpServer(binding.id, route.id, network);
```
We already used `bindingId` to resolve `binding` so at this point it is more correct to use `binding.id`.